amp.php

```diff
@@ -361,3 +363,52 @@ function amp_redirect_old_slug_to_new_url( $link ) {
 
 	return $link;
 }
+
+/**
+ * Hook into a comment submission if an AMP xhr post request.
+ */
+function amp_hook_comments_maybe() {
+
+	$method        = filter_input( INPUT_SERVER, 'REQUEST_METHOD', FILTER_SANITIZE_STRING );
+	$is_amp_submit = filter_input( INPUT_GET, '__amp_source_origin', FILTER_VALIDATE_URL );
+
+	if ( 'post' !== strtolower( $method ) || is_null( $is_amp_submit ) ) {
+		return;
+	}
+
+	// Add amp comment hooks.
+	add_filter( 'comment_post_redirect', 'amp_comment_post_handler', PHP_INT_MAX, 2 );
+	add_filter( 'wp_die_handler', 'amp_set_comment_submit_die_handler' );
+
+	// Send amp header.
+	header( 'AMP-Access-Control-Allow-Source-Origin: ' . $is_amp_submit, true );
+
+}
+
+/**
+ * Set the die handler on comment submit to output the error in json and send AMP headers.
+ */
+function amp_set_comment_submit_die_handler() {
+	return 'amp_send_error_json';
+}
+
+/**
+ * Send error creating message.
+ *
+ * @param string $message The error message to send.
+ */
+function amp_send_error_json( $message ) {
+	header( 'HTTP/1.1 400 BAD REQUEST' );
+	wp_send_json( array( 'error' => strip_tags( $message, 'strong' ) ) );
```

Comment:
@westonruter I used striptags because `amp-mustache` takes the text from element so it doesn't go deeper. meaning that anything within child elements are totally removed. So `<p><strong>ERROR</strong>:Comment is empty.</p>` becomes `:Comment is empty`

Reply:
Great to know. Thanks for sharing.